sale_advance_payment/models/sale.py

```diff
@@ -81,10 +83,16 @@ def _compute_advance_payment(self):
                     advance_amount += line_amount
             # Consider payments in related invoices.
             invoice_paid_amount = 0.0
-            for inv in order.invoice_ids:
-                invoice_paid_amount += (
-                    inv.amount_total_signed - inv.amount_residual_signed
-                )
+            for inv in order.invoice_ids.filtered(lambda x: x.state != "cancel"):
+                paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual
+                if inv.currency_id != order.currency_id:
+                    paid_amount = inv.currency_id._convert(
+                        paid_amount,
+                        order.currency_id,
+                        order.company_id,
+                        inv.invoice_date or inv.date,
+                    )
```

Comment:
To avoid computing things if not necessary
```suggestion
                    )
                    else:
                        paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual
```

Reply:
@rousseldenis I don't get your proposal, why should we have to add the 'else' if we have already set the paid amount before the 'if'?

Reply:
Because potentially, you can do this :

`paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual` even if `inv.currency_id != order.currency_id`

Reply:
but you need to do `paid_amount = inv.amount_total_in_currency_signed - inv.amount_residual` anyway in order to do the conversion when `inv.currency_id != order.currency_id`. If the currencies are not equal then you do the conversion, but if they are the same you already have the correct paid_amount. Not sure if I am missing something.

Reply:
Ooooh yes, didn't see, ok